From 75e28c7bada05ac7bd26d55b7a10f55bc5fda139 Mon Sep 17 00:00:00 2001
From: garypang13 <48883331+garypang13@users.noreply.github.com>
Date: Sun, 11 Oct 2020 18:04:50 +0800
Subject: [PATCH] Update smartdns

---
 package/openwrt/files/etc/init.d/smartdns | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/package/openwrt/files/etc/init.d/smartdns b/package/openwrt/files/etc/init.d/smartdns
index 2bf799e..e48b587 100644
--- a/package/openwrt/files/etc/init.d/smartdns
+++ b/package/openwrt/files/etc/init.d/smartdns
@@ -297,6 +297,12 @@ load_service()
 	config_get serve_expired "$section" "serve_expired" "0"
 	[ "$serve_expired" = "1" ] && conf_append "serve-expired" "yes"
 
+	config_get anti_ad "$section" "anti_ad" "0"
+	[ "$anti_ad" = "1" ] && {
+	wget -O $SMARTDNS_VAR_CONF_DIR/anti-ad-smartdns.conf https://anti-ad.net/anti-ad-for-smartdns.conf || true
+	conf_append "conf-file" "$SMARTDNS_VAR_CONF_DIR/anti-ad-smartdns.conf"
+	}
+
 	SMARTDNS_PORT="$port"
 
 	config_get cache_size "$section" "cache_size" ""
